src/Storefront/Controller/SearchController.php

```diff
@@ -41,6 +41,13 @@ public function search(SalesChannelContext $context, Request $request): Response
     {
         try {
             $page = $this->searchPageLoader->load($request, $context);
+            if($page->getListing()->getTotal() == 1)
+            {
+                $elements = $page->getListing()->getElements();
+                $product = array_pop($elements);
+                $productId = $product->getId();
+                return $this->forwardToRoute('frontend.detail.page', [], ['productId' => $productId]);
+            }
```

Comment:
Please test it though:

```suggestion
            if($page->getListing()->getTotal() === 1) {
                $product = $page->getListing()->getElements()->first();
                $productId = $product->getId();
                return $this->forwardToRoute('frontend.detail.page', [], ['productId' => $productId]);
            }
```